.github/workflows/reusable-build.yml

```diff
@@ -98,6 +98,25 @@ jobs:
       working-directory: ./${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}
       run: copy "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\Hostx64\x64\clang*" .
 
+    - name: Download demo repository artifacts
+      uses: dawidd6/action-download-artifact@v2.19.0
+      with:
```

Comment:
Why is this needed during the build and not during the test phase?

Reply:
I added it during the build phase so that the downloaded artifacts are then uploaded along with the build artifacts. Then:
1. The tests will directly get these when they download the build artifacts
2. If a test fails, the .o files used for the test will be available in the uploaded build artifacts.

Reply:
Suggest adding # comments in the file to explain.

Reply:
Added.